src/libs/export_metadata.c

```diff
@@ -0,0 +1,434 @@
+  GtkWidget *w = gtk_scrolled_window_new(NULL, NULL);
+  gtk_widget_set_size_request(w, DT_PIXEL_APPLY_DPI(300), DT_PIXEL_APPLY_DPI(300));
+  gtk_scrolled_window_set_policy(GTK_SCROLLED_WINDOW(w), GTK_POLICY_AUTOMATIC, GTK_POLICY_AUTOMATIC);
+  gtk_box_pack_start(GTK_BOX(vbox), w, TRUE, TRUE, 0);
+  GtkTreeView *view = GTK_TREE_VIEW(gtk_tree_view_new());
+  gtk_container_add(GTK_CONTAINER(w), GTK_WIDGET(view));
+  gtk_tree_view_set_headers_visible(view, FALSE);
+  gtk_widget_set_tooltip_text(GTK_WIDGET(view), _("list of available tags. click 'add' button or double-click on tag to add the selected one"));
+  gtk_tree_selection_set_mode(gtk_tree_view_get_selection(view), GTK_SELECTION_SINGLE);
+  GtkCellRenderer *renderer = gtk_cell_renderer_text_new();
+  GtkTreeViewColumn *col = gtk_tree_view_column_new_with_attributes("List", renderer, "text", 0, NULL);
+  gtk_tree_view_append_column(view, col);
+  GtkListStore *liststore = gtk_list_store_new(1, G_TYPE_STRING);
+  for(int i=0; i<dt_export_xmp_keys_n; i++)
+  {
+    GtkTreeIter iter;
+    gtk_list_store_append(liststore, &iter);
+    gtk_list_store_set(liststore, &iter, DT_LIB_EXPORT_METADATA_COL_XMP, dt_export_xmp_keys[i], -1);
+  }
+  gtk_tree_sortable_set_sort_column_id(GTK_TREE_SORTABLE(liststore), DT_LIB_EXPORT_METADATA_COL_XMP, GTK_SORT_ASCENDING);
+  gtk_tree_view_set_model(view, GTK_TREE_MODEL(liststore));
+  g_object_unref(liststore);
+  g_signal_connect(G_OBJECT(view), "button-press-event", G_CALLBACK(click_on_metadata_list), (gpointer)d);
+  
+  #ifdef GDK_WINDOWING_QUARTZ
+    dt_osx_disallow_fullscreen(dialog);
+  #endif
+    gtk_widget_show_all(dialog);
+  while (gtk_dialog_run(GTK_DIALOG(dialog)) == GTK_RESPONSE_YES)
+  {
+    add_selected_metadata(view, d);
+  }
+  gtk_widget_destroy(dialog);
+}
+
+static void remove_tag_from_list(dt_lib_export_metadata_t *d)
+{
+  GtkTreeIter iter;
+  GtkTreeModel *model = GTK_TREE_MODEL(d->liststore);
+  GtkTreeSelection *selection = gtk_tree_view_get_selection(d->view);
+  if(gtk_tree_selection_get_selected(selection, &model, &iter))
+  {
+    gtk_list_store_remove(d->liststore, &iter);
+  }
+}
+static void delete_tag_button_clicked(GtkButton *button, dt_lib_export_metadata_t *d)
+{
+  remove_tag_from_list(d);
+}
+
+static gboolean key_press_on_list(GtkWidget *widget, GdkEventKey *event, dt_lib_export_metadata_t *d)
+{
+  if(event->type == GDK_KEY_PRESS && event->keyval == GDK_KEY_Delete && !event->state)
+  {
+    remove_tag_from_list(d);
+    return TRUE;
+  }
+  return FALSE;
+}
+
+static void formula_edited(GtkCellRenderer *renderer, gchar *path, gchar *new_text, dt_lib_export_metadata_t *d)
+{
+  GtkTreeIter iter;
+  if (gtk_tree_model_get_iter_from_string(GTK_TREE_MODEL(d->liststore), &iter, path))
+    gtk_list_store_set(d->liststore, &iter, DT_LIB_EXPORT_METADATA_COL_FORMULA, new_text, -1);
+}
+
+uint32_t dt_lib_export_metadata_default_flags()
+{
+  const uint32_t flags = DT_META_EXIF | DT_META_METADATA | DT_META_GEOTAG | DT_META_TAG;
```

Comment:
DT_META_DT_HISTORY should be added there. It is the current default and this is really a safe setting.

Reply:
> this is really a safe setting

Do you think that really ?

This may be a code beauty but for me this is even not a nice to have ... people should be instructed to backup properly their databases instead of relying on this. The same for xmp files used as safety. At least I have the setting to avoid this last one but I'll have to think to remove this one every time . :-(
I'll do as you request. :-)

Reply:
Fact is I do use (and know other people doing that) this as a simple way to show to others the was it was developed. Giving the RAW + Jpeg with the result people can replay the development if needed. Otherwise you also have to hand the .xmp.

I had also one time looked at an exported jpg to recover an history stack I had lost.

Reply:
ok, I understand. And in any case I totally agree we shouldn't change the basic behavior without agreement of the dev team ... :-)